src/main/java/com/sap/cloudfoundry/client/facade/rest/CloudControllerRestClientImpl.java

```diff
@@ -317,80 +333,140 @@ public void bindServiceInstance(String applicationName, String serviceInstanceNa
+        return Lifecycle.builder()
+                        .type(LifecycleType.DOCKER)
+                        .data(DockerData.builder()
+                                        .build())
+                        .build();
+    }
+
+    private ApplicationRelationships buildApplicationRelationships() {
+        return ApplicationRelationships.builder()
+                                       .space(buildToOneRelationship(getTargetSpaceGuid()))
+                                       .build();
+    }
+
+    private GetApplicationProcessResponse getApplicationProcess(UUID applicationGuid) {
+        return delegate.applicationsV3()
+                       .getProcess(GetApplicationProcessRequest.builder()
+                                                               .type("web")
+                                                               .applicationId(applicationGuid.toString())
+                                                               .build())
+                       .block();
+    }
+
+    private void updateApplicationProcess(Staging staging, GetApplicationProcessResponse applicationProcess) {
+        if (staging.isSshEnabled() != null) {
+            updateSsh(staging.isSshEnabled());
+        }
+        UpdateProcessRequest.Builder updateProcessRequestBuilder = UpdateProcessRequest.builder()
+                                                                                       .processId(applicationProcess.getId())
+                                                                                       .command(staging.getCommand());
+        if (staging.getHealthCheckType() != null) {
+            updateProcessRequestBuilder.healthCheck(buildHealthCheck(staging));
+        }
+        delegate.processes()
+                .update(updateProcessRequestBuilder.build())
+                .block();
+    }
+
+    private void updateSsh(boolean isSshEnabled) {
+        delegate.featureFlags()
+                .set(SetFeatureFlagRequest.builder()
+                                          .name("ssh")
+                                          .enabled(isSshEnabled)
+                                          .build())
                 .block();
     }
 
+    private HealthCheck buildHealthCheck(Staging staging) {
+        HealthCheckType healthCheckType = parseHealthCheckType(staging.getHealthCheckType());
+        return HealthCheck.builder()
+                          .type(healthCheckType)
+                          .data(Data.builder()
+                                    .endpoint(staging.getHealthCheckHttpEndpoint())
+                                    .timeout(staging.getHealthCheckTimeout())
+                                    .build())
+                          .build();
+    }
+
+    private HealthCheckType parseHealthCheckType(String healthCheckType) {
+        // TODO: Deprecate the usage of "NONE" health check type since it is no longer supported by the CC v3 (PROCESS should be used
+        // instead)
```

Comment:
Then, can we validate the `Staging` object doesn't have "NONE" as it's health check type and either fail if it does or change it to a default value?

Reply:
Modified in multiapps-controller. Now it shows a warning otherwise it would be a breaking change